src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -0,0 +1,708 @@
+                .forEach(study -> study.getBrAPIObject().setLocationDbId(location.getLocationDbId()));
+    }
+
+    private void updateTrialDbId(BrAPITrial trial, String programKey) {
+        this.studyByNameNoScope.values().stream()
+                .filter(study -> study.getBrAPIObject().getTrialName().equals(Utilities.removeProgramKey(trial.getTrialName(), programKey ) ) )
+                .forEach(study -> study.getBrAPIObject().setTrialDbId(trial.getTrialDbId()));
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    private ArrayList<BrAPIGermplasm> getGermplasmByAccessionNumber(
+            List<String> germplasmAccessionNumbers,
+            UUID programId) throws ApiException {
+        List<BrAPIGermplasm> germplasmList = brAPIGermplasmDAO.getGermplasm(programId);
+        ArrayList<BrAPIGermplasm> resultGermplasm = new ArrayList<>();
+        // Search for accession number matches
+        for (BrAPIGermplasm germplasm: germplasmList) {
+            for (String accessionNumber: germplasmAccessionNumbers) {
+                if (germplasm.getAccessionNumber().equals(accessionNumber)) {
+                    resultGermplasm.add(germplasm);
+                    break;
+                }
+            }
+        }
+        return resultGermplasm;
+    }
+
+    private Map<String, PendingImportObject<BrAPIGermplasm>> initialize_existingGermplasmByGID(Program program, List<ExperimentObservation> experimentImportRows) {
+        Map<String, PendingImportObject<BrAPIGermplasm>> existingGermplasmByGID = new HashMap<>();
+        List<String> uniqueGermplasmGIDs = experimentImportRows.stream()
+                .map(ExperimentObservation::getGid)
+                .distinct()
+                .collect(Collectors.toList());
+
+        List<BrAPIGermplasm> existingGermplasms;
+        try {
+            existingGermplasms = this.getGermplasmByAccessionNumber(uniqueGermplasmGIDs, program.getId());
+            existingGermplasms.forEach(existingGermplasm -> existingGermplasmByGID.put(existingGermplasm.getAccessionNumber(), new PendingImportObject<>(ImportObjectState.EXISTING, existingGermplasm)));
+            return existingGermplasmByGID;
+        } catch (ApiException e) {
+            // We shouldn't get an error back from our services. If we do, nothing the user can do about it
+            throw new InternalServerException(e.toString(), e);
+        }
+    }
+
+    private Map<String, PendingImportObject<BrAPIStudy>> initialize_studyByNameNoScope(Program program, List<ExperimentObservation> experimentImportRows) {
+        Map<String, PendingImportObject<BrAPIStudy>> studyByNameNoScope = new HashMap<>();
+        List<String> uniqueStudyNames = experimentImportRows.stream()
+                .map(ExperimentObservation::getEnv)
+                .distinct()
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
```

Comment:
I think you need to add the program key to the name like you did in the trials search or else you won't get any results.